### PR TITLE
check i2d_ASN1_TYPE return value

### PR DESCRIPTION
**Pull request by bonniegong**
Add a length check to the return value of function i2d_ASN1_TYPE. 
Return an error instead of trying to malloc a negative number.
A similar check has already been implemented to the other usage of this function.

##### Checklist

- [ ] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

File: crypto/asn1/a_strex.c [hunk 1]
```diff
@@ -280,6 +280,8 @@ static int do_dump(unsigned long lflags, char_io *io_ch, void *arg,
     t.type = str->type;
     t.value.ptr = (char *)str;
     der_len = i2d_ASN1_TYPE(&t, NULL);
+    if (der_len < 0)
```
Comment (slontis) [inline crypto/asn1/a_strex.c:283]: <= maybe?

Comment (bonniegong) [reply crypto/asn1/a_strex.c:283]: i thought that malloc(0) would return a pointer that can be passed to free() without causing an error, but yes <= is better.

**Review by t8m — COMMENTED** [review]
This would qualify for CLA: trivial. So if you want to submit this under the CLA: trivial, please amend the commit message body to put `CLA: trivial` on a separate line there.

Otherwise please submit a CLA according to https://www.openssl.org/policies/cla.html

**Comment by bonniegong** [comment]
@t8m commit message has been amended, thank you for reminding.

File: crypto/asn1/a_strex.c (hunk 1 above)
Comment (paulidale) [reply crypto/asn1/a_strex.c]: The result of `malloc(0)` can always be passed to `free`.  I agree `<=` is better here.

**Review by paulidale — APPROVED** [review]

**Review by t8m — APPROVED** [review]
I agree with CLA: trivial

**Comment by t8m** [comment]
Should be applicable to 1.1.1. @paulidale do you agree with CLA: trivial and with the cherry pick to 1.1.1?

**Comment by paulidale** [comment]
Merged to master and 1.1.1, sorry for the delay.

**Pull request closed without merging**